test-lit/lower/pre-fir-tree01.f90

```diff
@@ -0,0 +1,129 @@
+! RUN: %f18 -fdebug-pre-fir-tree -fparse-only %s | FileCheck %s
+
+! Test structure of the Pre-FIR tree
+
+! CHECK: Subroutine foo
+subroutine foo()
+  ! CHECK: <<DoConstruct>>
+  ! CHECK: NonLabelDoStmt
+  do i=1,5
+    ! CHECK: PrintStmt
+    print *, "hey"
+    ! CHECK: <<DoConstruct>>
+    do j=1,5
```

Comment:
Missed a check for dostmt?

Reply:
Yes, thanks. Note that I did not always add checks for construct If/Do statements when I was only testing other statements that I put inside. But I have fixed it here since there is a check for the end statements.